src/snowflake/snowpark/mock/_connection.py

```diff
@@ -207,7 +207,7 @@ def wrap(*args, **kwargs):
             return log_and_telemetry
 
     def __init__(self, options: Optional[Dict[str, Any]] = None) -> None:
-        self._conn = Mock()
+        self._conn = Mock(expired=False)
```

Comment:
We should consider removing all these unrelated attributes instead of using MagicMock objects, this issue is an example of buggy behavior silenced by mock objects.

Reply:
I agreed, logged a JIRA to follow: https://snowflakecomputing.atlassian.net/browse/SNOW-1331149